src/panels/config/entities/entity-registry-settings-editor.ts

```diff
@@ -1194,20 +1198,18 @@ export class EntityRegistrySettingsEditor extends LitElement {
     ) {
       // Change invert setting
       const origEntityId = this.entry.options?.switch_as_x?.entity_id;
-      // remove current helper
-      await deleteConfigEntry(this.hass, this.helperConfigEntry.entry_id);
-
       if (!origEntityId) {
         // should not happen, guard for types
       } else {
-        const configFlow = await createConfigFlow(this.hass, "switch_as_x");
-        const configFlowResult = (await handleConfigFlowStep(
+        const configFlow = await createOptionsFlow(
+          this.hass,
+          this.helperConfigEntry.entry_id
+        );
+        const configFlowResult = (await handleOptionsFlowStep(
           this.hass,
           configFlow.flow_id,
           {
-            entity_id: origEntityId,
             invert: this._switchAsInvert,
-            target_domain: this._switchAsDomain,
           }
         )) as DataEntryFlowStepCreateEntry;
         if (configFlowResult.result?.entry_id) {
```

Comment:
We don't have to do this in this case? As the entity id stays the same there is no need to open a different more info dialog?